lib/jsonapi_compliable/adapters/abstract.rb

```diff
@@ -277,17 +260,17 @@ def disassociate(parent, child, association_name, association_type)
         raise 'you must override #disassociate in an adapter subclass'
       end
 
-      # This module gets mixed in to Sideload classes
-      # This is where you define methods like has_many,
-      # belongs_to etc that wrap the lower-level Sideload#allow_sideload
-      #
-      # @see Resource#allow_sideload
-      # @see Sideload#allow_sideload
-      # @see Adapters::ActiveRecord#sideloading_module
-      # @see Adapters::ActiveRecordSideloading
-      # @return the module to mix in
-      def sideloading_module
-        Module.new
+      # You want to override this!
+      # Map of association_type => sideload_class
+      # e.g.
+      # { has_many: Adapters::ActiveRecord::HasManySideload }
+      def sideloading_classes
+        {
+          has_many: ::JsonapiCompliable::Sideload::HasMany,
+          belongs_to: ::JsonapiCompliable::Sideload::BelongsTo,
+          has_one: ::JsonapiCompliable::Sideload::HasOne,
+          many_to_many: ::JsonapiCompliable::Sideload::ManyToMany
+        }
       end
```

Comment:
Instead of `sideloading_module`, these are all individual subclasses of `Sideload`. Just provide a mapping of relationship type > class.